### DISCUSSION TITLE
MutlipleLines outputs instead of Single Line for same timestamp for OPCUA tags

### DISCUSSION THREAD

**Discussion by ilyas032019** [gopcua/opcua#623] [Q&A] [answered]
Hi Team,
Currently i'm using this plugin to fetch data from OPCUAclients. 
My observation is on the design from, for each tag(from the opcua taglist configured in conf file) this plugin outputs the data as individual lines for each tag even if its receive data at same exact timestamps.

the issue comes when i use this telegraf plugin in QuestDB. I configured 10 OPCUA tag and i see 10 rows with same timestamps in the table out of which 9 are NULL and only 1 have values.

**Actual QuestDB table output:**
"timestamp","tag1","tag2","tag3","tag4","tag5","tag6","tag7","tag8","tag9","tag10"
"2022-12-10T23:59:28.000000Z",18.58072,,,,,,,,,
"2022-12-10T23:59:28.000000Z",,4.30858,,,,,,,,
"2022-12-10T23:59:28.000000Z",,,1155.558,,,,,,,
"2022-12-10T23:59:28.000000Z",,,,1411.614,,,,,,
"2022-12-10T23:59:28.000000Z",,,,,238.6185,,,,,
"2022-12-10T23:59:28.000000Z",,,,,,1280.09,,,,
"2022-12-10T23:59:28.000000Z",,,,,,,800.7842,,,
"2022-12-10T23:59:28.000000Z",,,,,,,,47.16882,,
"2022-12-10T23:59:28.000000Z",,,,,,,,,38.81365,
"2022-12-10T23:59:28.000000Z",,,,,,,,,,18.58072

**Expected QuestDB table output:: **One line output for same timestamp**. If value not present then NULL is obviuos output.**
"timestamp","tag1","tag2","tag3","tag4","tag5","tag6","tag7","tag8","tag9","tag10"
"2022-12-10T23:59:28.000000Z",18.58072, 4.30, 1155.5, 1411.6, 238.61, 1280.09, 800.78, 47.16882, 38.81, 18.58

Can we change this please. This might improve the throughput as well.

Thanks

**Comment by magiconair** [comment, marked as answer]
I think you better ask this in the community for the database you are using. This is just the code for the underlying protocol library.
